### PR TITLE
Define  jobs for community.yang collection

### PR DESCRIPTION
**Pull request by GomathiselviS**
Signed-off-by: GomathiselviS <gosriniv@redhat.com>

### CODE REVIEW

**Comment by GomathiselviS** [comment]
recheck

File: zuul.d/jobs.yaml [hunk 1]
```diff
@@ -1787,6 +1787,69 @@
...
+
+- job:
+    name: ansible-test-units-yang-python38
+    parent: ansible-test-units-base-python38
+    required-projects:
+      - name: github.com/ansible-collections/community.yang
+    vars:
+      ansible_collections_repo: github.com/ansible-collections/community.yang
+
+- job:
```
Comment (pabelanger) [inline zuul.d/jobs.yaml:1839]: lets drop integration jobs to start, this way we can properly test them before merging

File: zuul.d/project-templates.yaml [hunk 2]
```diff
@@ -289,12 +289,14 @@
         - build-ansible-collection:
             required-projects:
               - name: github.com/ansible-collections/ansible.netcommon
+              - name: github.com/ansible-collections/community.yang
```
Comment (pabelanger) [inline zuul.d/project-templates.yaml:292]: this should not be needed

File: zuul.d/project-templates.yaml [hunk 3]
```diff
@@ -289,12 +289,14 @@
...
     gate:
       queue: integrated
       jobs:
         - build-ansible-collection:
             required-projects:
               - name: github.com/ansible-collections/ansible.netcommon
+              - name: github.com/ansible-collections/community.yang
```
Comment (pabelanger) [inline zuul.d/project-templates.yaml:299]: same

File: zuul.d/project-templates.yaml [hunk 4]
```diff
@@ -442,6 +444,7 @@
...
```
Comment (pabelanger) [inline zuul.d/project-templates.yaml:447]: same

File: zuul.d/project-templates.yaml [hunk 5]
```diff
@@ -458,6 +461,7 @@
               - name: github.com/ansible-collections/ansible.netcommon
               - name: github.com/ansible-collections/cisco.iosxr
               - name: github.com/ansible-collections/junipernetworks.junos
+              - name: github.com/ansible-collections/community.yang
```
Comment (pabelanger) [inline zuul.d/project-templates.yaml:464]: same

File: zuul.d/project-templates.yaml [hunk 6]
```diff
@@ -578,6 +582,46 @@
       queue: integrated
       jobs: *ansible-collections-ansible-netcommon-units-jobs
 
+- project-template:
```
Comment (pabelanger) [inline zuul.d/project-templates.yaml:585]: lets move this into another PR

**Comment by pabelanger** [comment]
We can test this by adding .zuul.yaml file into yang collection and depends-on this PR.

File: zuul.d/jobs.yaml [hunk 7]
```diff
@@ -1787,6 +1787,69 @@
...
+    name: ansible-test-network-integration-community-yang-iosxr-netconf-python36
+    parent: ansible-test-network-integration-iosxr-python36
+    vars:
+      ansible_collections_repo: github.com/ansible-collections/community.yang
+      ansible_test_integration_targets: "yang_.*"
```
Comment (ganeshrn) [inline zuul.d/jobs.yaml:1844]: The prefix pattern is valid in case the collection has test for multiple platforms or in case of stable-2.9 ansible. Since all the test are contained in yang collection this prefix is not required. This can be replaced with `ansible_test_integration_targets: ""`

File: zuul.d/jobs.yaml [hunk 8]
```diff
@@ -1787,6 +1787,69 @@
...
+
+- job:
+    name: ansible-test-network-integration-community-yang-junos-vsrx-netconf-python36
+    parent: ansible-test-network-integration-junos-vsrx-python36
+    vars:
+      ansible_collections_repo: github.com/ansible-collections/community.yang
+      ansible_test_integration_targets: "yang_.*"
```
Comment (ganeshrn) [inline zuul.d/jobs.yaml:1851]: same as above

File: zuul.d/jobs.yaml (hunk 7 above)
Comment (ganeshrn) [reply zuul.d/jobs.yaml:1844]: To run all the test within yang collection a better way is to add `yang` group in iosxr and junos inventory file template refer
1) https://github.com/ansible/ansible-zuul-jobs/blob/1586f4e2aad2a211c7c7b196a937fd26d8c86abd/roles/ansible-test-inventory/templates/inventory-iosxr.j2#L7 

2) https://github.com/ansible/ansible-zuul-jobs/blob/1586f4e2aad2a211c7c7b196a937fd26d8c86abd/roles/ansible-test-inventory/templates/inventory-junos.j2#L7

By adding the group aliases in the yang collection integration test
Refer https://github.com/ansible-collections/community.yang/blob/af0f2da3faf465742274d99aca969954e2d3b79c/tests/integration/targets/fetch/aliases#L1
renaming test folder to prefix with `yang_*` is not required

File: zuul.d/project-templates.yaml [hunk 9]
```diff
@@ -578,6 +578,54 @@
...
+
+- project-template:
+    name: ansible-collections-community-yang-juniper-junos
+    check:
+      jobs: &ansible-collections-juniper-junos-yang-jobs
+        - ansible-test-network-integration-community-yang-junos-vsrx-netconf-python36:
+            vars:
+              ansible_test_collections: true
+        - build-ansible-collection:
+            required-projects:
```
Comment (pabelanger) [inline zuul.d/project-templates.yaml:604]: we should add ansible.netcommon here, it is a yang dependency

File: zuul.d/project-templates.yaml [hunk 10]
```diff
@@ -578,6 +578,54 @@
...
+
+- project-template:
+    name: ansible-collections-community-yang-cisco-iosxr
+    check:
+      jobs: &ansible-collections-cisco-iosxr-yang-jobs
+        - ansible-test-network-integration-community-yang-iosxr-netconf-python36:
+            vars:
+              ansible_test_collections: true
+        - build-ansible-collection:
+            required-projects:
```
Comment (pabelanger) [inline zuul.d/project-templates.yaml:620]: we should add ansible.netcommon here, it is a yang dependency
